dpnp/dpnp_iface_linearalgebra.py

```diff
@@ -274,58 +290,328 @@ def matmul(x1, x2, out=None, **kwargs):
     >>> c = np.ones([9, 5, 4, 3])
     >>> np.matmul(a, c).shape
     (9, 5, 7, 3)
+
     >>> a = np.array([[1, 0], [0, 1]])
     >>> b = np.array([[4, 1], [2, 2]])
     >>> np.matmul(a, b)
     array([[4, 1],
            [2, 2]])
 
```

Comment:
Please add descriptions for each example
```suggestion
For 2-D arrays it is the matrix product:

```